akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -55,10 +55,23 @@ private[http] object HttpServerBluePrint {
       }.withDeploy(Deploy.local)
     }, errorMsg = "Http.serverLayer is currently not reusable. You need to create a new instance for each materialization.")
 
+    def establishAbsoluteUri(requestOutput: RequestOutput): RequestOutput = requestOutput match {
+      case start: RequestStart ⇒
+        try {
+          val effectiveUri = HttpRequest.effectiveUri(start.uri, start.headers, securedConnection = false, defaultHostHeader)
+          start.copy(uri = effectiveUri)
+        } catch {
+          case e: IllegalUriException ⇒
+            MessageStartError(StatusCodes.BadRequest, ErrorInfo("Request is missing required `Host` header", e.getMessage))
```

Comment:
> if you haven't defined a default host, it could mean that you just don't want to support HTTP/1.0

Or it could mean that you simply don't know about this setting. ;) A quick note a la "in order to support HTTP/1.0 you have to specify..." wouldn't hurt I guess.

IMO the msg doesn't help the poor 1.0 client since it doesn't know about host headers. ;)